pkg/controller/pools.go

```diff
@@ -135,3 +141,31 @@ func (c *Controller) restartInitializedPool(ctx context.Context, tenant *miniov2
 
 	return nil
 }
+
+// waitUntilPoolPodAnnotated restarts a pool that is assumed to have been initialized
+func (c *Controller) waitUntilPoolPodAnnotated(ctx context.Context, tenant *miniov2.Tenant) (err error) {
+	tryCount := 0
+	var podList *corev1.PodList
+	for tryCount == 0 || (tryCount < 10 && err != nil) {
+		tryCount++
+		time.Sleep(time.Second * 2)
+		podList, err = c.kubeClientSet.CoreV1().Pods(tenant.Namespace).List(ctx, metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", miniov2.TenantLabel, tenant.Name),
+		})
+		if err != nil {
+			klog.Warning("Could not validate state of statefulset for pool", err)
+		}
+		generationMatch := true
+		for _, pod := range podList.Items {
+			if pod.Status.Phase == corev1.PodRunning && pod.Annotations[common.AnnotationsEnvTenantGeneration] != fmt.Sprintf("%d", tenant.Generation) {
+				generationMatch = false
+			}
+		}
+		if generationMatch {
+			break
+		}
+		err = errors.New("could not find pods with the correct generation")
```

Comment:
If we're allocating new errors, then we may as well be specifying what we expect (could make debugging easier).
```suggestion
		err = fmt.Errorf("Not all pods are generation %d", tenant.Generation)
```